src/main/scala/vexiiriscv/misc/IndirectCsrPlugin.scala

```diff
@@ -0,0 +1,57 @@
+package vexiiriscv.misc
+
+import spinal.core.{Bool, _}
+import spinal.core.fiber.Retainer
+import spinal.lib._
+import spinal.lib.fsm._
+import spinal.lib.misc.pipeline.Payload
+import spinal.lib.misc.plugin.FiberPlugin
+import vexiiriscv.Global._
+import vexiiriscv.execute.{CsrAccessPlugin, CsrCondFilter, CsrListFilter, CsrRamPlugin, CsrRamService}
+import vexiiriscv.riscv._
+import vexiiriscv.riscv.Riscv._
+import vexiiriscv._
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+class IndirectCsrPlugin(val withSupervisor : Boolean) extends FiberPlugin {
+  val logic = during setup new Area {
+    val cap = host[CsrAccessPlugin]
+    val buildBefore = retains(cap.csrLock)
+
+    awaitBuild()
+
+    val harts = for (hartId <- 0 until HART_COUNT) yield new Area {
+      val api = cap.hart(hartId)
+
+      val m = new Area {
+        val iselect = RegInit(U(0, XLEN bits))
+        api.readWrite(iselect, CSR.MISELECT)
```

Comment:
To ilustrate why plugins += new IndirectCsrPlugin(privParam.withSupervisor) should be optional (and disable by default)

```scala
        val iselect = RegInit(U(0, XLEN bits))
        api.readWrite(iselect, CSR.MISELECT)
``` 

is expensive